app/assets/javascripts/discourse/app/components/reviewable-item.js

```diff
@@ -225,6 +228,33 @@ export default Component.extend({
     this._penalize("showSilenceModal", reviewable, performAction);
   },
 
+  async clientEdit(reviewable, performAction) {
+    const post = await this.store.find("post", reviewable.post_id);
+    const topic_json = await Topic.find(post.topic_id, {});
+
+    const topic = Topic.create(topic_json);
+    post.set("topic", topic);
+
+    if (!this.currentUser) {
+      return this.dialog.alert(I18n.t("post.controls.edit_anonymous"));
+    } else if (!post.can_edit) {
```

Comment:
Could not this check be moved to the beginning of the function?

```js
async clientEdit(reviewable, performAction) {
    if (!this.currentUser) {
      return this.dialog.alert(I18n.t("post.controls.edit_anonymous"));
    }
    const post = await this.store.find("post", 
 reviewable.post_id);
    if (!post.can_edit) return false;

   /// rest...
```

Reply:
Yeah that's fine to do, I've updated